src/bootstrap/config.rs

```diff
@@ -1910,11 +1939,28 @@ impl Config {
 
         // Look for a version to compare to based on the current commit.
         // Only commits merged by bors will have CI artifacts.
+
+        // If we are running in CI, the current commit will never have artifacts already built.
+        let tip_commit = if CiEnv::is_ci() { "HEAD^" } else { "HEAD" };
+        // NOTE: we intentionally don't filter compiler/ and library/ here, even though we ideally
+        // would to avoid unnecessary downloads. The reason for that is complicated:
+        // - When bors creates a merge commit for a rollup, there are two levels of indirection:
+        //   1. The merge by bors
+        //   2. The merge from each individual commit in the rollup
+        // - Because of the indirection, `git log --author=bors compiler/` *only* includes merges that weren't from a rollup.
```

Comment:
I think this comment would be clearer if it more clearly distinguished between the "Rollup merge of ..." and "Auto merge of ..." commits for rollups. As-is I'm having trouble parsing this out.